todoapp.py

```diff
@@ -0,0 +1,153 @@
+import sqlite3
+
+# from collections import namedtuple
+
+from flask import Flask, request, render_template, jsonify
+from sqlite3 import Error
+
+
+# Row = namedtuple('Row', 'id, title, category, expired_date, '
+#                         'is_done, completed_date')
+
+app = Flask(__name__)
+# DB = 'todo.db'
+DB = 'test.db'
+
+
+@app.route('/')
+@app.route('/home/')
+def home():
+    return render_template('home.html', title='Home')
+
+
+@app.route('/api/tasks/', methods=['GET'])
+def show_tasks():
+    args = request.args
+    if args:
+        conditions = [f'{key} = "{args[key]}"' for key in args]
+        conditions = f'WHERE {" AND ".join(conditions)}'
+    else:
+        conditions = ''
+    fields = 'id, title, category, expired_date, is_done, completed_date'
+    query = f'SELECT {fields} FROM task {conditions}'
+    # print(query)
+
+    con = sqlite3.connect(DB)
+    cur = con.cursor()
+    try:
+        cur.execute(query)
+        rows = cur.fetchall()
+    except Error as e:
+        # print(e)
+        return f'Error <{e}> occurred during select data from database', 400
+    finally:
+        cur.close()
+        con.close()
+
+    if not rows:
+        return "Not found tasks according query", 404
+
+    tasks = []
+    for row in rows:
+        tasks.append(dict(zip(fields.split(', '), row)))
+    # return render_template('tasks.html', tasks=tasks, fields=fields.split(', '))
+    return jsonify(tasks)
+
+
+@app.route('/api/tasks/', methods=['POST'])
+def add_task():
+    recv_data = request.json
+    new_task = [recv_data.get('id'),
+                recv_data.get('title'),
+                recv_data.get('category'),
+                recv_data.get('expired_date'),
+                recv_data.get('is_done', 'underway'),
+                recv_data.get('completed_date', '')
+                ]
+    # print(new_task)
+
+    con = sqlite3.connect(DB)
```

Comment:
Like the next step on the future, we can put some logic for working with database in a separate class to avoiding duplicate code (Separate class, using context managers or decorators)

Reply:
To be honest i ddin't quite understand what you meant. But i tried to use the connection with database as a context manager (there was some problem with cursor and `fetchall()` and `fetchone()` functions, but i resolve it using iteration over the result of query execution).

Reply:
Ok, you could just move the logic for working with database in functions, [example](https://www.sqlitetutorial.net/sqlite-python/insert/). And use these functions inside of your API handlers.